conans/model/build_info.py

```diff
@@ -52,10 +54,16 @@ def _filter_paths(self, paths):
         abs_paths = [os.path.join(self.rootpath, p)
                      if not os.path.isabs(p) else p for p in paths]
         if self.filter_empty:
-            return [p for p in abs_paths if os.path.isdir(p)]
+            return [p for p in abs_paths if os.path.exists(p)]
```

Comment:
This change could be a bit risky

Reply:
Create a `_filter_file_paths` and keep this unmodified.